src/Types/Exception/UnknownColumnType.php

```
@@ -5,24 +5,45 @@
 namespace Doctrine\DBAL\Types\Exception;
 
 use Exception;
+use Throwable;
 
 use function sprintf;
 
 final class UnknownColumnType extends Exception implements TypesException
 {
-    public static function new(string $name): self
-    {
-        return new self(
-            sprintf(
-                'Unknown column type "%s" requested. Any Doctrine type that you use has '
+    private function __construct(
+        private string $requestedType,
+        ?string $context,
+        int $code = 0,
+        ?Throwable $throwable = null,
+    ) {
```

Changing the constructor signature or the constructor's visibility is a breaking change because we did not declare the constructor as internal. The named constructor `new()` took care of rendering the message which was fine. Please keep it that way. If you need a different message, feel free to add a second named constructor.

---

That sounds like a good approach. I've refactored it to a new method. I am not sure if the ensuing code duplication is something that could be easily factored out or if it is okay.

---

And a question about the tests. I couldn't find any tests for the abstract schema manager. Could you give me a lead on where I would need to add this?

---

What about `SchemaManagerFunctionalTestCase`?

---

Okay, I tried adding a test case but the problem is that it already fails on creation of the schema and it doesn't get to listing it. So when creating a column with an invalid type it throws an exception (but without context, as it doesn't know the table context yet).

I've added a test case in the column editor that it should throw an "Unknown column type" exception, is that enough or do I need to dig in how to create an invalid schema without the `Table::editor()`? Just to have it crash later when listing the schema.

---

> I've added a test case in the column editor that it should throw an "Unknown column type" exception

That test would pass without your modifications already, wouldn't it? This basically means that your changes to the schema manager are still not covered.

---

Correct, this test already would have worked without my changes. 

The problem I see is the test would fail when creating the column. Before I can assign it to any table to trigger the extended error message.